test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -7382,6 +7382,33 @@ BOOST_AUTO_TEST_CASE(inline_assembly_function_access)
 	BOOST_CHECK(callContractFunction("x()") == encodeArgs(u256(10)));
 }
 
+BOOST_AUTO_TEST_CASE(inline_assembly_labels_with_stack_info)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(uint y) {
+				assembly {
+					y 7
+					jump(fun)
+					exit[-1]:
+					y add
+					0 mstore
+					return(0, 0x20)
+					fun:
+					{
+						entry[a, b]:
+						a := div(a, b)
+						pop
```

Comment:
The programmer will have a hard time to tell that. Because of that, pop should never be used except by automated tools or for tiny tasks.